src/commands/inputs.rs

```diff
@@ -0,0 +1,540 @@
+                        }
+                    }
+
+                    return Some((
+                        namespace.push(name),
+                        Value::Array(values.into_iter().map(|(_, v)| v).collect()),
+                    ));
+                }
+                LiteralExpr::Map(m) => {
+                    let mut map = Map::new();
+
+                    for item in m.items() {
+                        let (k, v) = item.key_value();
+                        let key_name: String = match k {
+                            Expr::Literal(LiteralExpr::String(k)) => {
+                                let text = k.text().expect("should have string text");
+                                let mut buffer = String::new();
+                                text.unescape_to(&mut buffer);
+                                buffer
+                            }
+                            _ => k.text().to_string(),
+                        };
+
+                        if let Some((_key, value)) =
+                            self.expression(ty.clone(), Key::empty(), name, &v)
+                        {
+                            map.insert(key_name, value);
+                        }
+                    }
+
+                    return Some((namespace.push(name), Value::Object(map)));
+                }
+                LiteralExpr::Pair(p) => {
+                    let mut map = Map::new();
+                    let (left, right) = p.exprs();
+
+                    if let Some((_key, value)) =
+                        self.expression(ty.clone(), Key::empty(), name, &left)
+                    {
+                        map.insert("left".to_string(), value);
+                    }
+
+                    if let Some((_key, value)) =
+                        self.expression(ty.clone(), Key::empty(), name, &right)
+                    {
+                        map.insert("right".to_string(), value);
+                    }
+
+                    return Some((namespace.push(name), Value::Object(map)));
+                }
+                LiteralExpr::Object(v) => {
+                    let map = v
+                        .items()
+                        .filter_map(|item| {
+                            let (name, value) = item.name_value();
+                            self.expression(ty.clone(), Key::empty(), name.text(), &value)
+                        })
+                        .map(|(k, v)| (k.join().expect("key to join"), v))
+                        .collect::<Map<_, _>>();
+
+                    return Some((namespace.push(name), Value::Object(map)));
+                }
+                LiteralExpr::None(_) => {
+                    return Some((namespace.push(name), Value::Null));
+                }
+                LiteralExpr::Hints(_) | LiteralExpr::Input(_) | LiteralExpr::Output(_) => {
+                    // These are not allowed in the input section.
+                    panic!("unexpected literal expression");
```

Comment:
I think that's it's discouraged to call `panic!` directly in Rust? This could instead call `unreachable!()` which is clearer in intention